src/main/kotlin/org/jitsi/jibri/service/impl/FileRecordingJibriService.kt

```diff
@@ -118,11 +120,21 @@ class FileRecordingJibriService(private val fileRecordingParams: FileRecordingPa
     private var processMonitorTask: ScheduledFuture<*>? = null
 
     init {
-        sink = FileSink(fileRecordingParams.recordingDirectory, fileRecordingParams.callParams.callUrlInfo.callName)
+        sink = FileSink(
+            fileRecordingParams.recordingDirectory,
+            fileRecordingParams.callParams.callUrlInfo.callName
+        )
         jibriSelenium.addStatusHandler(this::publishStatus)
     }
 
     override fun start(): Boolean {
+        if (!createIfDoesNotExist(fileRecordingParams.recordingDirectory, logger)) {
+            return false
+        }
+        if (!Files.isWritable(fileRecordingParams.recordingDirectory)) {
+            logger.error("Unable to write to ${fileRecordingParams.recordingDirectory}")
+            return false
+        }
```

Comment:
Another corner case: I think if the path exists, but it is a file and it is writable we will proceed and try to treat it as a directory.